packages/graphql-client/src/types.ts

```diff
@@ -0,0 +1,59 @@
+export type CustomFetchAPI = (
+  url: string,
+  init?: {
+    method?: string;
+    headers?: HeadersInit;
+    body?: string;
+  }
+) => Promise<Response>;
+
+export interface OperationVariables {
+  [key: string]: any;
+}
+
+export interface Headers {
+  [key: string]: string;
+}
```

Comment:
I didn't use it here because we don't expect devs to pass in an actual `Header` object - just a set of key value pairs for their desired headers so we can easily combine any override headers if they are passed in to the `fetch` or `request` calls. 

I can rename it to something like `HeaderValues` if that's more clear - thoughts?

Reply:
I believe I had used `HeadersInit` but I remember when I was chatting with the hydrogen team, they had issues with it. Also, taking in an actual `Headers` instance won't allow for me to do the spread when I combine init headers and override headers. 

I'll stick with this implementation for now and if we see a need or usefulness to also include `Header` instances we can update then.

Reply:
Yes, agreed! I think `HeadersInit` won't be available everywhere so it's less risky to just use a plain object here.